src/application.coffee

```diff
@@ -220,13 +218,12 @@ fetch = (app, { deltaSource, setDeviceUpdateState = true } = {}) ->
 				if conf['RESIN_SUPERVISOR_DELTA'] == '1'
 					logSystemEvent(logTypes.downloadAppDelta, app)
 					deltaOpts = {
-						uuid, apiKey
+						uuid, apiKey, deltaSource
 						# use user-defined timeouts, but fallback to defaults if none is provided.
 						requestTimeout: checkInt(conf['RESIN_SUPERVISOR_DELTA_REQUEST_TIMEOUT'], positive: true)
-						applyTimeout: checkInt(conf['RESIN_SUPERVISOR_DELTA_APPLY_TIMEOUT'], positive: true) ? DEFAULT_DELTA_APPLY_TIMEOUT
+						applyTimeout: checkInt(conf['RESIN_SUPERVISOR_DELTA_APPLY_TIMEOUT'], positive: true)
```

Comment:
If deltas now track rsync, is the applyTimeout still relevant or should it be removed altogether?

Reply:
Now it applies as a timeout over the whole update (much like it was with the old deltas), and by default is 0 (due to falling back to using docker-delta's default) which means "no timeout". It'll be handy for any use case where there's need to limit the allowed time to update for whatever, but can't think of concrete use cases TBH. I'd be hesitant to remove it though.

Reply:
Yeah, true, I guess it's good to have it as a plan B in any case